### PR TITLE
Add initial contribution documentation

### LINKED ISSUE #48: Add some contribution documentation
It could help :beers:

### PR DESCRIPTION
**Pull request by lwm**
Saw you folks were missing this one. Here's a start!

Closes #48.

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+.. _`fork`: https://github.com/Yelp/undebt#fork-destination-box
+
+You'll then want to get a local copy of the code base:
+
+.. code-block:: bash
+
+    git clone git@github.com:<your-username>/undebt.git
+
+Getting Setup
```
Comment (ajm188) [inline docs/source/contributing.rst]: We have `make` targets for each of these steps.

- `make venv`
  - sets up virtualenv
  - worth adding a reminder to source `venv/bin/activate`
- `make dev`
  - all of the above + requirements-dev.txt
- `make test`
  - all of the above + runs tests
  - worth noting that once `make dev` has been run and you are in the virtualenv, you can use `tox -e <my-env>` to run tests for just one python, etc

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Or, run an individual environment:
+
+.. code-block:: bash
+
+    tox -e py35
+
+Note. If you do not have the required dependencies for each Tox environment, you will receive an error.
+Avoid this by passing the `--skip-missing-interpreters` option.
+
+Adding Documentation
```
Comment (ajm188) [inline docs/source/contributing.rst]: Soon, we should add a note here that the CLI docs are generated automatically, so you don't have to update the usage when you change the CLI.

**Comment by evhub** [comment]
This won't automatically show up in `index.rst` unless you add it to the table of contents there.

**Comment by lwm** [comment]
> This won't automatically show up in index.rst unless you add it to the table of contents there.

It's there?

https://github.com/Yelp/undebt/pull/47/files#diff-8aa79b83ed86e48962cc94787fc5cb56R16

**Comment by evhub** [comment]
@lwm Ah, looks good--didn't see that.

**Comment by lwm** [comment]
bump! This is good to go or?

File: docs/source/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+It is highly recommended that you create a `virtual environment`_ before installing the project dependencies.
+
+.. _`virtual environment`: http://docs.python-guide.org/en/latest/dev/virtualenvs/
+
+Both can be achieved by running:
+
+.. code-block:: bash
+
+    make dev
```
Comment (ajm188) [inline docs/source/contributing.rst]: Can you add a note after this code block reminding people to source their virtualenv?

File: docs/source/contributing.rst [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+.. code-block:: bash
+
+    make test
+
+Or, run an individual environment:
+
+.. code-block:: bash
+
+    tox -e py35
```
Comment (ajm188) [inline docs/source/contributing.rst]: Can you add a note that this will (probably) only work if they are inside of the virtualenv?

**Comment by ajm188** [comment]
@lwm Pending my issues, I'm happy, but I'd like @evhub to give a stamp of approval as well before merging

File: docs/source/contributing.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    git clone git@github.com:<your-username>/undebt.git
+
+Getting Setup
+-------------
+
+It is highly recommended that you create a `virtual environment`_ before installing the project dependencies.
+
+.. _`virtual environment`: http://docs.python-guide.org/en/latest/dev/virtualenvs/
+
+Both can be achieved by running:
```
Comment (evhub) [inline docs/source/contributing.rst]: What is "both" referring to here? I only see one thing above that `make dev` does, and that's set up your virtual environment.

**Comment by evhub** [comment]
@ajm188 lgtm

File: docs/source/contributing.rst (hunk 5 above)
Comment (ajm188) [inline docs/source/contributing.rst]: I think "both" == "creating virtualenv" + "install project deps"

But I agree this could probably be worded better.

Comment (ajm188) [inline docs/source/contributing.rst]: s/better/more clearly

File: docs/source/contributing.rst (hunk 3 above)
Comment (lwm) [inline docs/source/contributing.rst]: It's [already covered](https://github.com/Yelp/undebt/blob/master/Makefile#L7-L9)?

**Comment by ajm188** [comment]
lgtm. merging

**Comment by ajm188** [comment]
thanks @lwm!

**Pull request merged**